alpaca_trade_api/rest.py

```diff
@@ -612,29 +612,38 @@ def _data_get(self,
                   symbol_or_symbols: Union[str, List[str]],
                   api_version: str = 'v2',
                   endpoint_base: str = 'stocks',
+                  resp_grouped_by_symbol: bool = None,
                   **kwargs):
         page_token = None
         total_items = 0
         limit = kwargs.get('limit')
+        page_limit = DATA_V2_MAX_LIMIT
+        if endpoint_base == "news":
+            page_limit = NEWS_MAX_LIMIT
```

Comment:
[minor] again, I would add this logic to the caller instead.

Reply:
added `page_limit` as a parameter to `_data_get()` and passing `NEWS_MAX_LIMIT` from the caller. 
Let me know if this is not what you had in mind

Reply:
Yes, this is exactly what I had in mind.